### ISSUE TITLE
The name of the "is_greedy_or_beam_and_bucket" variable does not seem to match the logic

### ISSUE THREAD

**Issue by Jing1Ling** [huggingface/optimum-habana#1166]
### System Info

```shell
1.16.0
```

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

According to the comments [here](https://github.com/huggingface/optimum-habana/blob/44455150db41118a6031b06e3e90ab3d257c8432/optimum/habana/transformers/generation/utils.py#L870), when `bucket_internal` is not enabled, the `seq_len` of `input_tensor` will be padded to `max_input+max_new_tokens`. 

I enabled '`--bucket_internal`', and I think `input_ids.shape` should be (bs:128, max_input_tokens:128), but it is still padded. 
From what I've observed, the name of `is_greedy_or_beam_and_bucket` variable does not correspond to [the implemented logic](https://github.com/huggingface/optimum-habana/blob/44455150db41118a6031b06e3e90ab3d257c8432/optimum/habana/transformers/generation/utils.py#L835-L842).

```bash
python run_generation.py --model_name_or_path mosaicml/mpt-7b --use_hpu_graphs --use_kv_cache --limit_hpu_graph --batch_size 128  --max_input_tokens 128 --max_new_tokens 128 --trim_logits --attn_softmax_bf16 --warmup 3 --n_iterations 1 --bf16 --bucket_internal --bucket_size 64
```

I know there is no bucket internal implementation of the MPT model in the main branch yet, but the above implementations are all in 'run_generation.py' file.

### Expected behavior

I think `input_ids.shape` should be (bs:128, max_input_tokens:128), not (128, 256)

**Comment by regisss** [comment]
Cc @ssarkar2

**Comment by Jing1Ling** [comment]
I mistakenly thought `bucket_internal` was bucket. After carefully reading the code, I understand that `bucket_internal` refers to enabling bucket inside the model. Bucket is enabled when `bucket_size>0`, but when` bucket_internal` is enabled, bucket outside the model is no longer executed and padding is still performed on the input.
